ui_framework/components/button/button.js

```diff
@@ -0,0 +1,147 @@
+import React, {
+  PropTypes,
+} from 'react';
+
+import classNames from 'classnames';
+import keyMirror from 'keymirror';
+
+import { KuiButtonIcon } from './button_icon/button_icon';
+
+const commonPropTypes = {
+  type: PropTypes.string,
```

Comment:
I prefer not having this `KuiButton.TYPE.BASIC` thing, and rather just do

```
type: PropTypes.oneOf(['basic', 'hollow', 'danger', 'primary'])
```

I just think this reads _so_ much better in code:

```
<KuiButton type='danger' />
```

It'll be as safe as the constant, plus if we for some reason want to rename things later on, that's super-simple to fix using jscodeshift.